## Riddler Express

[Link](https://fivethirtyeight.com/features/the-little-mathematically-determined-house-on-the-prairie/)

To solve this problem, I calculated a transition matrix where each state is how many umbrellas are at home.

The value in each element in the matrix $M_{ij}$ is the probability of transitioning between state $i$ and state $j$ while staying dry.

In [2]:
import numpy as np
AM = np.array([
    [0.5, 0.5, 0. , 0. ],
    [0. , 0.5, 0.5, 0. ],
    [0. , 0. , 0.5, 0.5],
    [0. , 0. , 0. , 0.5]
])

PM = np.array([
    [0.6, 0. , 0. , 0. ],
    [0.4, 0.6, 0. , 0. ],
    [0. , 0.4, 0.6, 0. ],
    [0. , 0. , 0.4, 0.6]
])

This shows how different states transition after the walk to the office in the morning.

There is a 50% chance that the number of umbrellas stays the same , i.e. the state doesn't change, and a 50% the number of umbrellas goes down by one except if there are no umbrellas. The magnitude of the final state is the probability that Louie stays dry after the transition. Here you can see the probability of staying dry is 100% unless there are no umbrellas at home, then it is 50%.

In [4]:
print "N  Initial       After Morning"

for umbrellas_at_home in range(4):
    state = np.zeros(4)
    state[umbrellas_at_home] = 1
    print umbrellas_at_home, state, np.matmul(AM, state)

N  Initial       After Morning
0 [1. 0. 0. 0.] [0.5 0.  0.  0. ]
1 [0. 1. 0. 0.] [0.5 0.5 0.  0. ]
2 [0. 0. 1. 0.] [0.  0.5 0.5 0. ]
3 [0. 0. 0. 1.] [0.  0.  0.5 0.5]


So we start with the state where 2 umbrellas are at home and multiply by $PM*AM$ 5 times. The sum of the final state should be the probability of staying dry.

In [5]:
state = np.array([0,0,1,0])

day = np.matmul(PM,AM)
week = np.linalg.matrix_power(day,5)
endstate = np.matmul(week,state)

sum(endstate), (1-sum(endstate))*5*2

(0.6927, 3.0730000000000004)

Louie has a 69.27% chance of staying dry after 5 days. So he will be rained on about 3 times.